tests/NIRCam_ecfs/S3_NIRCam.ecf

```diff
@@ -1,5 +1,6 @@
 # Eureka! Control File for Stage 3: Data Reduction
 
+inst        nircam      # Only needed during testing since we haven't run S2
```

Comment:
This shouldn't need to be added - the pipeline should be able to run from jwst-produced S2 files. We haven't needed it so far - where is the new requirement coming from?

Reply:
This comes from Line 120 in s3_reduce.py.
```
if meta.inst == 'miri':
    meta.xwindow[0] *= meta.expand
    meta.xwindow[1] *= meta.expand
else:
    meta.ywindow[0] *= meta.expand
    meta.ywindow[1] *= meta.expand
```
`meta.inst` gets set when running Stage 2 of Eureka!, but never gets set in Stage 3.

Reply:
I don't see an obvious workaround because `meta = util.readfiles(meta, log)` can't get called until the log file is defined, but that doesn't happen until we're inside the double for loop and we don't want to repeatedly expand xwindow and ywindow each time we iterate.

Reply:
> This comes from Line 120 in s3_reduce.py.
> 
> ```
> if meta.inst == 'miri':
>     meta.xwindow[0] *= meta.expand
>     meta.xwindow[1] *= meta.expand
> else:
>     meta.ywindow[0] *= meta.expand
>     meta.ywindow[1] *= meta.expand
> ```
> 
> `meta.inst` gets set when running Stage 2 of Eureka!, but never gets set in Stage 3. This means we can't currently load in JWST-produced Stage 2 files without setting `inst` in the ECF.

Why can't these lines just be moved to the `inst.read` functions? I see no reason to have them in s3_reduce.py. There is already code in the miri.py file that only gets run for the first batch, so we already have the capability to do special things the first time through the read function